### ISSUE TITLE
Selection from a popup feature geometrynot working

### ISSUE THREAD

**Issue by gioman** [3liz/lizmap-javascript-scripts#70]
I believe that my test project is ok, configured as explaine in the script readme. Using QGIS Server/Desktop 3.28.15 and LMWC 3.6.10

When clicking on the icon/star in the feature popup (to trigger the selection on the other layers) the dev console shows this:

```
Uncaught TypeError: lizMap.mainLizmap.digitizing.drawLayer.getSource is not a function
    popupSelectionButtonClickHandler getMedia:86
    addSelectionButton getMedia:32
    lizmappopupdisplayed getMedia:178
    lizmappopupdisplayed getMedia:154
    triggerEvent OpenLayers.js:255
    Z map.js:2679
    getfeatureinfo map.js:3440
    triggerEvent OpenLayers.js:255
    triggerGetFeatureInfo OpenLayers.js:1692
    handleResponse OpenLayers.js:1693
    callback OpenLayers.js:1689
    bind OpenLayers.js:77
    runCallbacks OpenLayers.js:1563
    onreadystatechange OpenLayers.js:1559
    dispatchEvent OpenLayers.js:1593
    fReadyStateChange OpenLayers.js:1596
    onreadystatechange OpenLayers.js:1579
    open OpenLayers.js:1574
    issue OpenLayers.js:1558
    GET OpenLayers.js:1564
    request OpenLayers.js:1691
    getInfoForClick OpenLayers.js:1683
    callback OpenLayers.js:718
    delayedCall OpenLayers.js:1384
    bind OpenLayers.js:77
    setTimeout handler*queuePotentialClick OpenLayers.js:1380
    handleSingle OpenLayers.js:1380
    click OpenLayers.js:1377
    triggerEvent OpenLayers.js:255
    handleBrowserEvent OpenLayers.js:261
    bindAsEventListener OpenLayers.js:77
    observe OpenLayers.js:238
    attachToElement OpenLayers.js:247
    initialize OpenLayers.js:244
    initialize OpenLayers.js:300
    init map.js:1362
    init map.js:6282
[getMedia:86:44](https://demo.naturalgis.pt/lizmap/www/index.php/view/media/getMedia?repository=qgisserverdemos&project=teste&mtime=1709052737&path=media%2Fjs%2Fdefault%2FselectionFromPopupGeometry.js)

​
```

**Comment by altheaFeu** [comment]
Hi gioman,
Your error indicates that `getSource` does not exist for your object. This method doesn't work with LMWC 3.6 but works with LMWC 3.7. I believe this is because version 3.7 removed code from OpenLayers 2.
You can test `lizMap.mainLizmap.digitizing.drawLayer.getSource().clear()` in the JS console for both of these projects to see the difference:

[LMWC 3.6 Projects]([https://demo.snap.lizmap.com/lizmap_3_6/index.php](https://demo.snap.lizmap.com/lizmap_3_6/index.php)) 
[LMWC 3.7 Project](https://demo.snap.lizmap.com/lizmap_3_7/index.php/view/embed?repository=modules&project=pgmetadata#3.857582,43.609135,3.886894,43.618969%7Ctrees,footways,water_surfaces,buildings,gardens%7Cdefault,default,default,default,default%7C1,1,1,1,1)

**Comment by gioman** [comment]
> Your error indicates that `getSource` does not exist for your object. This method doesn't work with LMWC 3.6 but works with LMWC 3.7. I believe this is because version 3.7 removed code from OpenLayers 2.

@altheaFeu thanks for the reply. The script is 2 years old, with no recent changes, so does not seems to be the case(?).



> You can test `lizMap.mainLizmap.digitizing.drawLayer.getSource().clear()` in the JS console for both of these projects to see the difference:
> 
> [LMWC 3.6 Projects](https://demo.snap.lizmap.com/lizmap_3_6/index.php) [LMWC 3.7 Project](https://demo.snap.lizmap.com/lizmap_3_7/index.php/view/embed?repository=modules&project=pgmetadata#3.857582,43.609135,3.886894,43.618969%7Ctrees,footways,water_surfaces,buildings,gardens%7Cdefault,default,default,default,default%7C1,1,1,1,1)

unclear to me what I should test. The first link does not point to a specific project, the second one points to a project that has no selection tool available.

**Comment by gioman** [comment]
> I expressed myself poorly. If you type `lizMap.mainLizmap.digitizing.drawLayer.getSource().clear()` in the JS console of [this project](https://demo.snap.lizmap.com/lizmap_3_6/index.php/view/map?repository=features&project=cats) (LMWC 3.6), you will encounter an error,

@altheaFeu but the js code is 2 years old with no changes in the meantime, this means that it has been created as it is long before LM 3.7.

**Comment by altheaFeu** [comment]
@gioman Indeed, in this case, I wouldn't be able to tell where the error comes from.

**Comment by gioman** [comment]
@altheaFeu thanks anyway!

**Comment by altheaFeu** [comment]
Hi @gioman 
I needed this tool for my own project, so I rectified the script.
I also translated the native JavaScript functions into jQuery.
You can find the correction [here](https://github.com/3liz/lizmap-javascript-scripts/pull/78), and it successfully operated on my project with LMWC 3.5.10.

**Comment by gioman** [comment]
@altheaFeu thanks!

**Comment by Gustry** [comment]
> You can find the correction https://github.com/3liz/lizmap-javascript-scripts/pull/78, and it successfully operated on my project with LMWC 3.5.10.

Thanks @altheaFeu for all contributions ! Very appreciated. I didn't get time last week to review (I was in Grenoble for the QGIS user conf where I met some of your company :) )

Sorry, but did you have a instance with LWC 3.7 running ?
For now, Javascript files do not have metadata about which LWC versions they are targeting. I'm bit reluctant to merge files if it's targeting LWC 3.5, as this version is already not maintained by us.

Of course, if your files are on API that didn't change in LWC 3.7, there is no issue, but I will need to ask my colleagues for the review (@nboisteault mainly)

**Comment by gioman** [comment]
If the PR makes the script work on LM 3.6 is well worth the merge, many are holding on the move to LMWC 3,7.
